skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,299 @@
+
+    def estimate(self, src, dst):
+        """Estimate optimal coefficients that describes the deformation of the points.
+
+
+        Parameters
+        ----------
+        src : (N, 2) array_like
+            Control point at source coordinates
+        dst : (N, 2) array_like
+            Control point at destination coordinates
+
+        Returns
+        -------
+        success: bool
+            True, indicates that the coefficients were successfully estimated.
+
+        Notes
+        -----
+        -  The number of source and destination points must match (N).
+        """
+
+        check_nD(src, 2)
+        check_nD(dst, 2)
+
+        if len(src) < 3 or len(dst) < 3:
+            raise ValueError(f"{src} points less than 3 is considered undefined.")
+
+        if src.shape != dst.shape:
+            raise ValueError("src and dst shape must be identical")
+
+        self.src = src
+        n, d = src.shape
+
+        dist = sp.spatial.distance.cdist(self.src, self.src)
+        K = _radial_basis_kernel(dist)
+        P = np.hstack([np.ones((n, 1)), src])
+        L = np.zeros((n + 3, n + 3), dtype=np.float32)
+        L[:n, :n] = K
+        L[:n, -3:] = P
+        L[-3:, :n] = P.T
+        V = np.concatenate([dst, np.zeros((d + 1, d))])
+        self.coefficients = np.dot(np.linalg.inv(L), V)
+        return True
+
+    def _radial_distance(self, x, y):
+        """Compute the radial distance"""
+        Pi_x = self.src[:, 0]
+        Pi_y = self.src[:, 1]
+
+        dx = Pi_x[:, np.newaxis] - x
+        dy = Pi_y[:, np.newaxis] - y
+        r = np.sqrt(dx**2 + dy**2)
+        radial_dist = _radial_basis_kernel(r)
+        return radial_dist
+
+    def _spline_function(self, x, y, radial_dist, coeffs):
+        """Solve the spline function in the X and Y directions"""
+        w = coeffs[:-3]
+        a1, ax, ay = coeffs[-3:]
+        summation = np.dot(w, radial_dist)
+        return a1 + ax * x + ay * y + summation
+
+
+def _radial_basis_kernel(r):
+    """Compute basis kernel for thine-plate splines.
+
+    Parameters
+    ----------
+    r : (4, N) ndarray
+        Input array representing the norm distance between interlandmark
+        distances for the source form and based on the (x,y) coordinates
+        for each of these points.
+
+    Returns
+    -------
+    U : (4, N) ndarray
+        Calculated kernel function U.
+    """
+    _small = 1e-8  # Small value to avoid divide-by-zero
+    r_sq = r**2
+    U = np.where(r == 0.0, 0.0, r_sq * np.log(r_sq + _small))
+    return U
+
+
+def tps_warp(
+    image, src, dst, *, output_region=None, interpolation_order=1, grid_scaling=1
+):
+    """Return an array of warped images.
+
+    Define a thin-plate-spline warping transform that warps from the
+    src to the dst, and then warp the given images by
+    that transform.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    src : (N, 2)
+        Control point at source coordinates.
```

Comment:
```suggestion
        Control points at source coordinates.
```
There are N control points, right?